### PR TITLE
Adding detailed comments for Amazon Review example

### PR DESCRIPTION
**Pull request by NivekT**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #271

Differential Revision: [D34562198](https://our.internmc.facebook.com/intern/diff/D34562198)

### CODE REVIEW

File: examples/text/amazonreviewpolarity.py [hunk 1]
```diff
@@ -31,19 +34,43 @@
...
+    # but the content of the DataPipe is unchanged. Therefore, `cache_compressed_dp` still contains a list of URL.
     cache_compressed_dp = url_dp.on_disk_cache(
         filepath_fn=lambda x: os.path.join(root, _PATH), hash_dict={os.path.join(root, _PATH): MD5}, hash_type="md5"
     )
+
+    # `GDriveReader` takes in URLs to GDrives files, and yields a tuple of file name and IO stream.
+    # After that, `.end_caching` saves the previous DataPipe's outputs onto the disk. In this case,
+    # the results from GDriveReader (i.e. the downloaded compressed archive) will be saved onto the disk.
+    # Upon saving the results, the DataPipe returns the paths to the cached files.
     cache_compressed_dp = GDriveReader(cache_compressed_dp).end_caching(mode="wb", same_filepath_fn=True)
```
Comment (ejguan) [inline examples/text/amazonreviewpolarity.py:53]: Super nit: We can split this line into two parts and comment on each op.
cache_compressed_dp = GDriveReader(cache_compressed_dp)
cache_compressed_dp = cache_compressed_dp.end_caching(mode="wb", same_filepath_fn=True)

File: examples/text/amazonreviewpolarity.py [hunk 2]
```diff
@@ -31,19 +34,43 @@
...
     )
+
+    # The follow step opens the cache files using `FileOpener`, loads the content of TAR files as a stream,
+    # and filter for the file name that we are looking for (either "train.csv" or "test.csv").
     cache_decompressed_dp = (
-        FileOpener(cache_decompressed_dp, mode="b").load_from_tar().filter(lambda x: _EXTRACTED_FILES[split] in x[0])
+        FileOpener(cache_decompressed_dp, mode="b")
+        .load_from_tar()
+        .filter(lambda fname_and_stream: _EXTRACTED_FILES[split] in fname_and_stream[0])
     )
```
Comment (ejguan) [inline examples/text/amazonreviewpolarity.py:67]: ditto

File: examples/text/amazonreviewpolarity.py [hunk 3]
```diff
@@ -31,19 +34,43 @@
...
+
+    # ".end_caching" saves the decompressed file onto disks and yields the path to the file.
     cache_decompressed_dp = cache_decompressed_dp.end_caching(mode="wb", same_filepath_fn=True)
 
+    # Open the decompressed file.
     data_dp = FileOpener(cache_decompressed_dp, mode="b")
+
+    # Finally, this parses content of the decompressed CSV file and returns the result line by line.
     return data_dp.parse_csv().map(fn=lambda t: (int(t[0]), " ".join(t[1:])))
```
Comment (ejguan) [inline examples/text/amazonreviewpolarity.py:76]: ditto.

**Review by ejguan — APPROVED** [review]

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34562198).

**Pull request closed without merging**